xdsl/parser.py

```diff
@@ -1094,6 +977,10 @@ def parse_operation(self) -> Operation:
         # Check for custom op format
         op_name = self.try_parse_bare_id()
         if op_name is not None:
+            assert isinstance(
+                self, XDSLParser
+            ), "Only xDSL format currently supports custom op parsing"
+            assert ret_types is not None, "Return types must be in xDSL format"
```

Comment:
Maybe this is github reviews being too local, but I don't quite get the error message here. This sounds like it always needs to be the case, which is not true, right? Is this only on an execution path that can be reached when parsing xDSL syntax?

Reply:
Yeah that's the case. both of the asserts are just putting in code the implicit assumptions. return types are printed before the op only in xDSL syntax, in MLIR they're at the end of the line.

Reply:
Okay, works for me :+1: